spec/min_heap_binary_tree_spec.rb

```diff
@@ -78,4 +78,396 @@
+
+      min_heap_tree.delete(node_5)
+
+      expect(min_heap_tree.tree).to eq(
+        [
+          nil,
+          initial_node,
+          node_7,
+          node_6,
+          node_9,
+          node_21,
+          node_8,
+          node_15,
+          node_17
+        ]
+      )
+    end
+
+    it "properly deletes a node with 2 children that are equal" do
+      initial_node = Node.new 1
+      min_heap_tree = MinHeapBinaryTree.new(initial_node)
+
+      node_5 = Node.new 5
+      node_6 = Node.new 6
+      node_9 = Node.new 9
+      node_11 = Node.new 11
+      node_8 = Node.new 8
+      node_15 = Node.new 15
+      node_17 = Node.new 17
+      node_21 = Node.new 21
+
+      min_heap_tree.insert(node_5)
+      min_heap_tree.insert(node_6)
+      min_heap_tree.insert(node_9)
+      min_heap_tree.insert(node_9)
+      min_heap_tree.insert(node_8)
+      min_heap_tree.insert(node_15)
+      min_heap_tree.insert(node_17)
+      min_heap_tree.insert(node_21)
+
+      min_heap_tree.delete(node_5)
+
+      expect(min_heap_tree.tree).to eq(
+        [
+          nil,
+          initial_node,
+          node_9,
+          node_6,
+          node_9,
+          node_21,
+          node_8,
+          node_15,
+          node_17
+        ]
+      )
+    end
+
+    it "properly deletes a node and swaps up if child is smaller then parent" do
+      initial_node = Node.new 1
+      min_heap_tree = MinHeapBinaryTree.new(initial_node)
+
+      node_9 = Node.new 9
+      node_19 = Node.new 19
+      node_17 = Node.new 17
+      node_11 = Node.new 11
+      node_22 = Node.new 22
+      node_27 = Node.new 27
+      node_21 = Node.new 21
+      node_33 = Node.new 33
+
+      min_heap_tree.insert(node_9)
+      min_heap_tree.insert(node_22)
+      min_heap_tree.insert(node_17)
+      min_heap_tree.insert(node_11)
+      min_heap_tree.insert(node_33)
+      min_heap_tree.insert(node_27)
+      min_heap_tree.insert(node_21)
+      min_heap_tree.insert(node_19)
+
+      min_heap_tree.delete(node_33)
+
+      expect(min_heap_tree.tree).to eq(
+        [
+          nil,
+          initial_node,
+          node_9,
+          node_19,
+          node_17,
+          node_11,
+          node_22,
+          node_27,
+          node_21
+        ]
+      )
+    end
+  end
+
+  # describe "#print" do
+  #   it "properly prints the whole tree" do
+  #     initial_node = Node.new 5
+  #     min_heap_tree = MinHeapBinaryTree.new(initial_node)
+  #
+  #     node_7 = Node.new 7
+  #     node_3 = Node.new 3
+  #     node_9 = Node.new 9
+  #     node_1 = Node.new 1
+  #     node_2 = Node.new 2
+  #     node_4 = Node.new 4
+  #     node_8 = Node.new 8
+  #
+  #     min_heap_tree.print
+  #
+  #     expect(min_heap_tree.tree).to eq(
+  #       [
+  #         nil,
+  #         node_3,
+  #         node_2,
+  #         node_8,
+  #         node_7,
+  #         initial_node,
+  #         node_4,
+  #         node_9
+  #       ]
+  #     )
+  #   end
+  # end
+
+  describe "#find" do
```

Comment:
We'll want a spec that tests the `false` case. Meaning trying to find a node that is not in the tree.